examples/apps/presence-tracker/src/MouseTracker.ts

```diff
@@ -18,98 +22,56 @@ export interface IMousePosition {
-		this.emit("mousePositionChanged");
-	};
+	private readonly cursor: LatestValueManager<IMousePosition>;
 
 	constructor(
-		public readonly audience: IAzureAudience,
-		private readonly signaler: ISignaler,
+		private readonly presence: IPresence,
+		// eslint-disable-next-line @typescript-eslint/ban-types
+		statesWorkspace: PresenceStates<{}>,
 	) {
 		super();
 
-		this.audience.on("memberRemoved", (clientId: string, member: IMember) => {
-			const clientIdMap = this.posMap.get(member.id);
-			if (clientIdMap !== undefined) {
-				clientIdMap.delete(clientId);
-				if (clientIdMap.size === 0) {
-					this.posMap.delete(member.id);
-				}
-			}
+		statesWorkspace.add("cursor", Latest({ x: 0, y: 0 }));
+		this.cursor = statesWorkspace.props.cursor;
+
+		this.presence.events.on("attendeeDisconnected", () => {
 			this.emit("mousePositionChanged");
 		});
 
-		this.signaler.on("error", (error) => {
-			this.emit("error", error);
+		this.cursor.events.on("updated", () => {
+			this.emit("mousePositionChanged");
 		});
-		this.signaler.onSignal(
-			MouseTracker.mouseSignalType,
-			(clientId: string, local: boolean, payload: IMouseSignalPayload) => {
-				this.onMouseSignalFn(clientId, payload);
-			},
-		);
+
 		window.addEventListener("mousemove", (e) => {
-			const position: IMousePosition = {
+			// Alert all connected clients that there has been a change to this client's mouse position
+			this.cursor.local = {
 				x: e.clientX,
 				y: e.clientY,
 			};
-			this.sendMouseSignal(position);
 		});
 	}
 
 	/**
-	 * Alert all connected clients that there has been a change to a client's mouse position
+	 * A map of session clients to mouse positions.
 	 */
-	private sendMouseSignal(position: IMousePosition) {
-		this.signaler.submitSignal(MouseTracker.mouseSignalType, {
-			userId: this.audience.getMyself()?.id,
-			pos: position,
-		});
-	}
+	public getMousePresences(): Map<ISessionClient, IMousePosition> {
+		const statuses: Map<ISessionClient, IMousePosition> = new Map();
 
-	public getMousePresences(): Map<string, IMousePosition> {
-		const statuses: Map<string, IMousePosition> = new Map<string, IMousePosition>();
-		this.audience.getMembers().forEach((member, userId) => {
-			member.connections.forEach((connection) => {
-				const position = this.getMousePresenceForUser(userId, connection.id);
-				if (position !== undefined) {
-					statuses.set(member.name, position);
-				}
-			});
-		});
+		for (const { client, value } of this.cursor.clientValues()) {
+			if (client.getConnectionStatus() === SessionClientStatus.Connected) {
+				statuses.set(client, value);
+			}
+		}
 		return statuses;
 	}
 
-	public getMousePresenceForUser(
-		userId: string,
-		clientId: string,
-	): IMousePosition | undefined {
-		return this.posMap.get(userId)?.get(clientId);
+	/**
+	 * Set the allowable latency for mouse cursor updates.
+	 *
+	 * @param latency - the maximum allowable latency for updates. Set to undefined to revert to the default value.
```

Comment:
Thank you for documenting the meaning of undefined!!! I so rarely see that when reviewing PRs, and it's nearly always really important info.